src/bud/index.ts

```diff
@@ -0,0 +1,60 @@
+import {api} from './api'
+import {hooks} from './hooks'
+import {util} from './util'
+import {plugin} from './plugin'
+import {state} from './state'
+import {mode} from './mode'
+
+import type {Hooks} from './hooks'
+import type {Mode} from './mode'
+import type {Plugin} from './plugin'
+import type {Util} from './util'
+import type {State} from './state'
+import type {
+  Alias,
+  Auto,
+  Babel,
+  Bundle,
+  Copy,
+  CopyAll,
+  Src,
+  SrcPath,
+  Sync,
+  Watch,
+} from './api'
+
+export type Bud = {
+  hooks: Hooks,
+  util: Util,
+  plugin: Plugin,
+  mode: Mode,
+  state: State | undefined,
+  alias: Alias,
+  auto: Auto,
+  babel: Babel,
+  bundle: Bundle,
+  copy: Copy,
+  copyAll: CopyAll,
+  src: Src,
+  srcPath: SrcPath
+  sync: Sync
+  watch: Watch
+}
+
+export const bud: Bud = {
```

Comment:
todo: finalize the api methods, loaded onto the exported `bud` object.

These can't be mapped or spread as it breaks intellisense. Gotta do it like this, I think (or use ES6 imports but unless someone else is taking that work on I'm leaving it as a nice-to-have for 3.0.0).

Reply:
done